### ISSUE TITLE
load_weights dont work in tf.keras: W tensorflow/core/util/tensor_slice_reader.cc:95] Could not open "mobilenet_model": Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?

### ISSUE THREAD

**Issue by kiflowb777** [tensorflow/tensorflow#34893]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 1.14, 1.15, 2.0, 2.1rc0
- Keras version: 2.2.4-tf and 2.2.4
- Python version: 3.6
- CUDA/cuDNN version: 10.0 and 10.1 / 7.6.5
- GPU model and memory: RTX 2060 6GB

When I use the **tf.keras** and try to load the weights from mobilenet modeI get an error"
`W tensorflow/core/util/tensor_slice_reader.cc:95] Could not open "mobilenet_model": Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?`

When use only **keras** it's ok.

**Simple code to reproduce the issue**
```
import os

use_tfkeras = True

if use_tfkeras:
    from tensorflow.compat.v1.keras.applications.mobilenet import MobileNet
else:
    from keras.applications.mobilenet import MobileNet


def save_mobilenet_weights(alpha, filename):
    mobilenet = MobileNet(alpha=alpha, input_tensor=None, include_top=False, weights='imagenet', pooling=None)
    if use_tfkeras:
        mobilenet.save_weights(filepath=os.path.abspath(filename), overwrite=True, save_format='h5')
    else:
        mobilenet.save_weights(filepath=os.path.abspath(filename), overwrite=True)

def load_mobilenet_weights(alpha, filename):
    mobilenet = MobileNet(alpha=alpha, input_tensor=None, include_top=False, weights=None, pooling=None)
    mobilenet.load_weights(os.path.abspath(filename))


alpha = 0.75
filename = 'mobilenet_model'

save_mobilenet_weights(alpha, filename)
load_mobilenet_weights(alpha, filename)
```

**Comment by kiflowb777** [comment]
Related topics:
https://github.com/tensorflow/tensorflow/issues/22947

**Comment by oanush** [comment]
@kiflowb777 ,
When `use_tfkeras = True` was able to run the code successfully on TF-2.1rc0,2.0,1.14, please fins the [gist](https://colab.sandbox.google.com/gist/oanush/acbfe415f78f6b3e746eefb11a3c27dc/34893.ipynb) of colab for your reference.Kindly gist of colab if the issue faced.Thanks!

**Comment by kiflowb777** [comment]
Please run "Runtime" -> "view runtime logs", the warning can be found there.

**Comment by oanush** [comment]
Issue replicating for given code.Thanks!

**Comment by sdnjly** [comment]
The problem seems solved when set the `save_weights_only=True` , like this:
`model_checkpoint = ModelCheckpoint(
            'best_model.ckpt',
            monitor='val_loss',
            save_best_only=True,
            save_weights_only=True)`

**Comment by saikumarchalla** [comment]
@kiflowb777  Could you please check with latest TF version and  let us know if the issue still persists. Thanks!

**Comment by kiflowb777** [comment]
Similar problems:
https://github.com/tensorflow/models/issues/2676
https://github.com/tensorflow/models/issues/2675
https://github.com/tensorflow/tensorflow/issues/7696
https://stackoverflow.com/questions/48438122/import-meta-graph-fails-with-data-loss-not-an-sstable-bad-magic-number
